src/main/scala/com/amazon/deequ/repository/sparktable/SparkMetricsRepository.scala

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ * use this file except in compliance with the License. A copy of the License
+ * is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+package com.amazon.deequ.repository.sparktable
+
+import com.amazon.deequ.analyzers.Analyzer
+import com.amazon.deequ.analyzers.runners.AnalyzerContext
+import com.amazon.deequ.metrics.Metric
+import com.amazon.deequ.repository._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Row, SaveMode, SparkSession}
+
+class SparkTableMetricsRepository(session: SparkSession, tableName: String) extends MetricsRepository {
+
+  private val SCHEMA = StructType(Array(
+    StructField("result_key", StringType),
+    StructField("metric_name", StringType),
+    StructField("metric_value", StringType),
+    StructField("result_timestamp", StringType),
+    StructField("serialized_context", StringType)
+  ))
+
+  override def save(resultKey: ResultKey, analyzerContext: AnalyzerContext): Unit = {
+    val serializedContext = AnalysisResultSerde.serialize(Seq(AnalysisResult(resultKey, analyzerContext)))
+
+    val rows = analyzerContext.metricMap.map { case (analyzer, metric) =>
+      Row(resultKey.toString, analyzer.toString, metric.value.toString,
+        resultKey.dataSetDate.toString, serializedContext)
+    }.toSeq
+
+    val metricDF = session.createDataFrame(session.sparkContext.parallelize(rows), SCHEMA)
```

Comment:
We can create a case class for the schema and use `toDF` in order to avoid having to create a sequence of rows and a DF using that.

```
case class SparkTableMetric(resultKey: String, metricName: String, metricValue: String)

import session.implicits._

val metricsDF = Seq(
  SparkTableMetric("key_1", "Completeness", 1.0),
  SparkTableMetric("key_2", "Uniqueness", 0.5),
).toDF()

```